### PR TITLE
doc/user: add a source troubleshooting guide

### PR DESCRIPTION
**Pull request by benesch**
Take @guswynn's excellent advice on how to use the
`mz_source_statistics` table, which I excised from https://github.com/MaterializeInc/materialize/pull/16937, and expand it
into a full-fledged source troubleshooting guide.

### Motivation

   * This PR adds documentation.

### Tips for reviewer

Still a bunch of TODOs to resolve, but I thought it'd be fun to iterate on this together! @guswynn I hope you're not too miffed that I pulled this out of #16937—I just feel like it's so much more natural to talk about these things inline with real usage examples of the `mz_internal.mz_source_statistics` table.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a
  companion cloud PR to account for those changes that is tagged with
  the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  - n/a

### CODE REVIEW

**Comment by benesch** [comment]
I'm unlikely to have time to push this much forward, but hopefully @guswynn you might?

**Comment by guswynn** [comment]
@benesch yeah I can take this!

**Comment by benesch** [comment]
Awesome, thank you so much, Gus!

**Comment by guswynn** [comment]
`bool_and`/`bool_or` are a can of works, so I gave the user a `count(*) FILTER` replacement for now: https://github.com/MaterializeInc/materialize/issues/17349

**Comment by benesch** [comment]
Just merged `bool_and` and `bool_or`, so this should be good to merge with those functions once v0.43 ships next Weds!

**Comment by guswynn** [comment]
Thanks @benesch ! I'll rebase

File: doc/user/content/ops/troubleshooting.md [hunk 1]
```diff
@@ -302,3 +302,160 @@ SELECT DISTINCT
...
+Repeatedly query the
+[`mz_source_statistics`](/sql/system-catalog/mz_internal/#mz_source_statistics)
+table and look for ingestion statistics that advance over time:
+
+```sql
+SELECT
+    SUM(bytes_received) AS bytes_received,
+    SUM(messages_received) AS messages_received,
+    SUM(updates_staged) AS updates_staged,
+    SUM(updates_committed) AS updates_committed,
```
Comment (chuck-alt-delete) [inline doc/user/content/ops/troubleshooting.md:362]: extra comma will cause an error

Comment (guswynn) [reply doc/user/content/ops/troubleshooting.md:362]: sql's biggest flaw :(

File: doc/user/content/ops/troubleshooting.md [hunk 2]
```diff
@@ -302,3 +302,160 @@ SELECT DISTINCT
...
+when you query the source, the source may still be ingesting its initial
+snapshot. See the next section.
+
+## Has my source ingested its initial snapshot?
+
+Query the `snapshot_comitted` field of the
+[`mz_source_statistics`](/sql/system-catalog/mz_internal/#mz_source_statistics) table:
+
+```sql
+SELECT bool_and(snapshot_committed) as snapshot_committed
```
Comment (chuck-alt-delete) [inline doc/user/content/ops/troubleshooting.md:338]: Materialize tells me:
```
ERROR:  bool_and not yet supported
```

Comment (chuck-alt-delete) [reply doc/user/content/ops/troubleshooting.md:338]: This works for me I think:

```sql
SELECT
    sources.id,
    sources.name,
    MIN(snapshot_committed::int)::bool as snapshot_committed
FROM
    mz_internal.mz_source_statistics stats
    JOIN mz_catalog.mz_sources sources
        ON sources.id = stats.id
GROUP BY sources.id, sources.name;
```

Comment (guswynn) [reply doc/user/content/ops/troubleshooting.md:338]: You have to be a recent main!

Comment (guswynn) [reply doc/user/content/ops/troubleshooting.md:338]: I don't think we want to promote `MIN` to do this!

Comment (guswynn) [reply doc/user/content/ops/troubleshooting.md:338]: We will merge this pr after the release

**Comment by benesch** [comment]
@guswynn @morsapaes — are we good to merge this?

**Comment by morsapaes** [comment]
Ah, this slipped my mind last week! Pushed a commit to patch some queries that weren't runnable and update the Support page to not mention "source-available Materialize".

**Comment by benesch** [comment]
Awesome, thank you! Merging!

**Pull request merged**